File: simpl_hiring_challenge_skillenza.com/skillenza_simpl_junior_engineer.cpp
#include <bits/stdc++.h>

using namespace std;

bool dist_possible(int num_boxes, int boxes[], int num_children)
{
    bool subset[num_boxes+1][num_children+1];

    for (int i = 0; i <= num_boxes; i++)
        subset[i][0] = true;

    for (int i = 1; i <= num_children; i++)
        subset[0][i] = false;

    for (int i = 1; i <= num_boxes; i++) {
        for (int j = 1; j <= num_children; j++)
        {
            if(j<boxes[i-1])
                subset[i][j] = subset[i-1][j];
            if (j >= boxes[i-1])
                subset[i][j] = subset[i-1][j] ||
                    subset[i - 1][j-boxes[i-1]];
        }
    }

    // uncomment this code to print table
    //for (int i = 0; i <= num_boxes; i++) {
        //for (int j = 0; j <= num_children; j++)
            //printf ("%4d", subset[i][j]);
        //printf("\n");
    //}

    return  subset[num_boxes][num_children];
}


int main() {
    int t = 0;
    int tmpint;
    int result;

    int num_boxes;
    int num_children;

    cin >> t;

    for(int i=0; i<t; i++) {
        cin >> num_boxes;
        int boxes[num_boxes];
        for(int b=0; b<num_boxes; b++) {
            cin >> tmpint;
            boxes[b] = tmpint;
        }
        cin >> num_children;


        //cout << num_children << "->";
        //for(int b=0; b<num_boxes; b++) cout << boxes[b] << " ";
        //cout << endl;

        result = dist_possible(num_boxes, boxes, num_children);
        if(result)
            cout << "YES" << endl;
        else
            cout << "NO" << endl;
    }

    return 0;
}

File: simpl_hiring_challenge_skillenza.com/skillenza_simpl_junior_engineer.py
def dist_possible_2(num_boxes, boxes, num_children):
    # print(num_boxes, boxes, num_children)
    psb = [[False for j in range(num_children + 1)] for i in range(num_boxes + 1)]

    for box in range(num_boxes + 1):
        psb[box][0] = True

    for i in range(num_boxes + 1):
        for j in range(num_children + 1):
            if j < boxes[i - 1]:
                psb[i][j] = psb[i - 1][j]
            else:
                psb[i][j] = psb[i - 1][j] or psb[i - 1][j - boxes[i - 1]]

    # for i in range(num_boxes + 1):
        # for j in range(num_children + 1):
            # print(psb[i][j], end=" ")
        # print()

    return psb[num_boxes][num_children]


def dist_possible(num_boxes, boxes, num_children):
    # print(num_boxes, boxes, num_children)
    if num_children == 0:
        return True
    elif num_children > 0 and num_boxes == 0:
        return False
    elif boxes[num_boxes - 1] > num_children:
        return dist_possible(num_boxes - 1, boxes, num_children)
    else:
        return dist_possible(num_boxes - 1, boxes, num_children) or dist_possible(num_boxes - 1, boxes, num_children - boxes[num_boxes - 1])


def solve():
    testcases = int(input())
    for _ in range(testcases):
        boxes = [int(x) for x in input().split(' ')]
        num_children = int(input())
        if sum(boxes[1:]) < num_children:
            result = False
        else:
            result = dist_possible(boxes[0], boxes[1:], num_children)
        # print(boxes[0], boxes[1:], num_children, result)
        print("YES" if result else "NO")

solve()
